subworkflows/local/trim_workflow/tests/main.nf.test

```diff
@@ -0,0 +1,85 @@
+nextflow_workflow {
+
+    name "Test Subworkflow QC_WORKFLOW"
+    script "../main.nf"
+    workflow "TRIM_WORKFLOW"
+    tag "qc"
+    tag "subworkflow"
+
+    test("TRIM_WORKFLOW - Homo sapiens - FASTQs - FastP true") {
+
+        when {
+
+            workflow {
+                """
+                // ch_reads
+                input[0] = Channel.of(
+                    [
+                        [id: "test_fastq"],
+                        [
+                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
+                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_2.fastq.gz", checkIfExists: true)
+                        ]
+                    ])
+                input[1] = Channel.empty()
+                input[2] = true
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success },
+                { with(workflow.out) {
+                    assert snapshot(
+                        versions.collect{ file(it) },
+                        ch_reads_all[0][1].collect { file(it) },
+                        ch_reads_fusioncatcher[0][1].collect { file(it) },
+                        file(ch_fastp_html[0][1]).name,
+                        file(ch_fastp_json[0][1]),
+                        ch_fastqc_trimmed[0][1].collect { file(it).name }
+                    ).match() }
+                }
+            )
+        }
+    }
+
+    test("TRIM_WORKFLOW - Homo sapiens - FASTQs - FastP false") {
+
+        when {
+
+            workflow {
+                """
+                // ch_reads
+                input[0] = Channel.of(
+                    [
+                        [id: "test_fastq"],
+                        [
+                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
+                            file("https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/modules/data/genomics/homo_sapiens/illumina/fastq/test_2.fastq.gz", checkIfExists: true)
+                        ]
+                    ])
+                input[1] = Channel.empty()
+                input[2] = false
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success },
+                { with(workflow.out) {
+                    assert snapshot(
+                        versions.size() == 0,
+                        ch_reads_all[0][1].size() == 2,
+                        ch_reads_fusioncatcher[0][1].size() == 2,
```

Comment:
I might be wrong here but if I understand this correctly, these 2 can be file-checked instead of size-checked?

Reply:
@atrigila When these files are not consumed by any process (like the second test), the paths still point to github and the `checkIfExists` condition fails. I seems to me that the github paths got converted to a local path. 
That's why I reduced it to checking the number of elements only.